build.gradle

```diff
@@ -107,6 +127,31 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
+String checkForNewEap (String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/'+intellijMajorVersion+'.][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?).pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    return matcher.group(1)
+  }
+}
```

Comment:
Hmm as IntelliJ points out here: not all execution paths return a value... pls return a `null` explicitly if link has not been found